packages/framework/src/Model/Payment/Service/PaymentServiceFacade.php

```diff
@@ -2,55 +2,59 @@
-use App\Model\Payment\Transaction\PaymentTransaction;
-use App\Model\Payment\Transaction\PaymentTransactionDataFactory;
-use App\Model\Payment\Transaction\PaymentTransactionFacade;
+namespace Shopsys\FrameworkBundle\Model\Payment\Service;
+
 use Psr\Log\LoggerInterface;
 use Shopsys\FrameworkBundle\Component\FlashMessage\FlashMessageTrait;
 use Shopsys\FrameworkBundle\Component\Money\Money;
+use Shopsys\FrameworkBundle\Model\GoPay\Exception\GoPayNotConfiguredException;
+use Shopsys\FrameworkBundle\Model\GoPay\Exception\GoPayPaymentDownloadException;
+use Shopsys\FrameworkBundle\Model\GoPay\GoPayFacade;
+use Shopsys\FrameworkBundle\Model\Order\Order;
+use Shopsys\FrameworkBundle\Model\Payment\Payment;
+use Shopsys\FrameworkBundle\Model\Payment\PaymentSetupCreationData;
+use Shopsys\FrameworkBundle\Model\Payment\PaymentSetupCreationDataFactory;
+use Shopsys\FrameworkBundle\Model\Payment\Service\Exception\PaymentServiceFacadeNotRegisteredException;
+use Shopsys\FrameworkBundle\Model\Payment\Transaction\Exception\PaymentTransactionHasNoAssignedPayment;
+use Shopsys\FrameworkBundle\Model\Payment\Transaction\PaymentTransaction;
+use Shopsys\FrameworkBundle\Model\Payment\Transaction\PaymentTransactionDataFactory;
+use Shopsys\FrameworkBundle\Model\Payment\Transaction\PaymentTransactionFacade;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 class PaymentServiceFacade
 {
     use FlashMessageTrait;
 
     /**
-     * @var \App\Model\Payment\Service\PaymentServiceInterface[]
+     * @var \Shopsys\FrameworkBundle\Model\Payment\Service\PaymentServiceInterface[]
      */
-    private array $paymentServices;
+    protected array $paymentServices;
 
     /**
-     * @param \App\Model\Payment\Transaction\PaymentTransactionFacade $paymentTransactionFacade
-     * @param \App\Model\Payment\Transaction\PaymentTransactionDataFactory $paymentTransactionDataFactory
-     * @param \App\Model\GoPay\GoPayFacade $goPayFacade
+     * @param \Shopsys\FrameworkBundle\Model\Payment\Transaction\PaymentTransactionFacade $paymentTransactionFacade
+     * @param \Shopsys\FrameworkBundle\Model\Payment\Transaction\PaymentTransactionDataFactory $paymentTransactionDataFactory
+     * @param \Shopsys\FrameworkBundle\Model\GoPay\GoPayFacade $goPayFacade
      * @param \Psr\Log\LoggerInterface $logger
      * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+     * @param \Shopsys\FrameworkBundle\Model\Payment\PaymentSetupCreationDataFactory $paymentSetupCreationDataFactory
      */
     public function __construct(
-        private readonly PaymentTransactionFacade $paymentTransactionFacade,
-        private readonly PaymentTransactionDataFactory $paymentTransactionDataFactory,
+        protected readonly PaymentTransactionFacade $paymentTransactionFacade,
+        protected readonly PaymentTransactionDataFactory $paymentTransactionDataFactory,
         GoPayFacade $goPayFacade,
```

Comment:
Why isn't it marked as 'protected'?

Reply:
because it is not a property - I am not able to use constructor property promotion here. The `GoPayFacade` is wired via the constructor and stored in `$paymentServices` array (instances of PaymentServiceInterface). If I get it correctly, the `PaymentServiceFacade` class is ready to handle more payment services thanks to that (If we had e.g. Sofort, we would just add `SofortFacade` here